### PR TITLE
Teach the server and client how to create new topics and schemas

### LINKED ISSUE #108: Support for explicit topic creation
Once we support schemas being enforced at the topic level, we'll need to support clients explicitly creating topics so that they can define a topic as conforming to a schema.

Things to do:
- [x] Support a new "create topic" protocol command, which also has  an optional schema. Default to "string" if not set.
- [x] Support creating new topics from the client repl. Maybe syntax like "create topic /foo with schema int32"

### PR DESCRIPTION
**Pull request by dburkart**
This PR includes the following:

* Logic in the protocol for creating new topics (and their associated schemas)
* A new "create topic" command in the REPL
* Plumbs logic through database internals for validating new schemas against existing topic hierarchies.

closes #108

### CODE REVIEW

**Comment by dburkart** [comment]
This also introduces a new list command, "list schemas", which returns a list of topics that have a non-string schema set.

**Comment by gideonw** [comment]
Should we add a new column to the query output for type? Or for list topics with schema?

**Comment by dburkart** [comment]
I think so. Either a column, or the client should indicate non-string data, I.e "int32(40)" or "binary(<hidden>)"

**Comment by dburkart** [comment]
I decided to go with putting the schema directly in the data for non-strings, since we'll have to elide it in some cases anyway (e.g. for binary)

**Comment by dburkart** [comment]
What this looks like now:

```
> query all
+-------------------------------------+---------------+------------+
|                TIME                 |     TOPIC     |    DATA    |
+-------------------------------------+---------------+------------+
| 2022-12-31T17:37:20.480499375-08:00 | /             | some data! |
| 2022-12-31T23:22:25.647325-08:00    | /sensors/temp | int32(32)  |
+-------------------------------------+---------------+------------+
```

File: pkg/proto/message.go [hunk 1]
```diff
@@ -564,3 +569,42 @@ func (rq *ListResponse) Unmarshal(b []byte) error {
...
+		return err
+	}
+	length := binary.BigEndian.Uint32(lengthPrefix)
+	topic := make([]byte, length)
+	m, err := io.ReadFull(buf, topic)
+	if err != nil {
+		return err
+	}
+	rq.Topic = string(topic)
+	rq.Schema = string(b[n+m:])
```
Comment (gideonw) [inline pkg/proto/message.go:605]: For some reason, I feel like this might error for empty strings.

File: pkg/proto/message_test.go [hunk 2]
```diff
@@ -323,3 +323,22 @@ func TestListResponse(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func TestCreateTopicRequest(t *testing.T) {
```
Comment (gideonw) [inline pkg/proto/message_test.go:327]: Add a case for no schema.

**Review by gideonw — CHANGES_REQUESTED** [review]
LGTM, one small change.

File: pkg/proto/message_test.go (hunk 2 above)
Comment (dburkart) [reply pkg/proto/message_test.go:327]: Added a test.

File: pkg/proto/message.go (hunk 1 above)
Comment (dburkart) [reply pkg/proto/message.go:605]: This appears to work, added test coverage to ensure it does though.

**Pull request merged**